### PR TITLE
dm: add error binlog filter description

### PR DESCRIPTION
**Pull request by GMHDBJD**
### First-time contributors' checklist 

- [ ] I've signed [**Contributor License Agreement**](https://cla-assistant.io/pingcap/docs-cn) that's required for repo owners to accept my contribution.

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions (in Chinese)](https://github.com/pingcap/docs-cn/blob/master/CONTRIBUTING.md#版本选择指南).

- [x] master (the latest development version)
- [x] v7.5 (TiDB 7.5 versions)
- [ ] v7.4 (TiDB 7.4 versions)
- [ ] v7.3 (TiDB 7.3 versions)
- [ ] v7.1 (TiDB 7.1 versions)
- [ ] v6.5 (TiDB 6.5 versions)
- [ ] v6.1 (TiDB 6.1 versions)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)
- [ ] v5.1 (TiDB 5.1 versions)
- [ ] v5.0 (TiDB 5.0 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

File: dm/dm-binlog-event-filter.md [hunk 1]
```diff
@@ -54,18 +55,40 @@ filters:
...
         - 不在该 rule 的 `events` 中。
         - 如果规则的 `sql-pattern` 不为空的话，对应的 SQL 没有匹配上 `sql-pattern` 中任意一项。
     - `Ignore`：黑名单。如果满足下面两个条件之一就会被过滤掉：
         - 在该 rule 的 `events` 中。
         - 如果规则的 `sql-pattern` 不为空的话，对应的 SQL 可以匹配上 `sql-pattern` 中任意一项。
-    - 同一个表匹配上多个规则时，将会顺序应用这些规则，并且黑名单的优先级高于白名单，即如果同时存在规则 `Ignore` 和 `Do` 应用在某个表上，那么 `Ignore` 生效。
+    - `Error`：报错名单。如果满足下面两个条件之一就会报错：
+        - 在该 rule 的 `events` 中。
+        - 如果规则的 `sql-pattern` 不为空的话，对应的 SQL 可以匹配上 `sql-pattern` 中任意一项。
+    - 同一个表匹配上多个规则时，将会顺序应用这些规则，并且黑名单的优先级高于报错名单高于白名单，即如果同时存在规则 `Ignore` 和 `Error` 应用在某个表上，那么 `Ignore` 生效；如果同时存在规则 `Error` 和 `Do` 应用在某个表上，那么 `Error` 生效。
```
Comment (hfxsd) [inline dm/dm-binlog-event-filter.md:91]:
```suggestion
    - 同一个表匹配上多个规则时，将会顺序应用这些规则，并且黑名单的优先级高于报错名单，报错名单的优先级高于白名单，即如果同时存在规则 `Ignore` 和 `Error` 应用在某个表上，那么 `Ignore` 生效；如果同时存在规则 `Error` 和 `Do` 应用在某个表上，那么 `Error` 生效。
```

File: dm/dm-binlog-event-filter.md [hunk 2]
```diff
@@ -5,7 +5,7 @@ summary: 了解 DM 的关键特性 binlog 事件过滤 (Binlog event filter) 的
 
 ## TiDB Data Migration Binlog 事件过滤
 
-TiDB Data Migration (DM) 的 Binlog 事件过滤 (Binlog event filter) 是比迁移表[黑白名单](/dm/dm-block-allow-table-lists.md)更加细粒度的过滤规则，可以指定只迁移或者过滤掉某些 `schema / table` 的指定类型 binlog，比如 `INSERT`、`TRUNCATE TABLE`。
+TiDB Data Migration (DM) 的 Binlog 事件过滤 (Binlog event filter) 是比迁移表[黑白名单](/dm/dm-block-allow-table-lists.md)更加细粒度的过滤规则，可以指定只迁移，过滤，或者报错某些 `schema / table` 的指定类型 binlog，比如 `INSERT`、`TRUNCATE TABLE`。
```
Comment (hfxsd) [inline dm/dm-binlog-event-filter.md:8]:
```suggestion
TiDB Data Migration (DM) 的 Binlog 事件过滤 (Binlog event filter) 是比迁移表[黑白名单](/dm/dm-block-allow-table-lists.md)更加细粒度的过滤规则，可以指定只迁移、过滤、或者报错某些 `schema / table` 的指定类型 binlog，比如 `INSERT` 和 `TRUNCATE TABLE`。
```

File: dm/dm-binlog-event-filter.md [hunk 3]
```diff
@@ -54,18 +55,40 @@ filters:
     | rename table    | DDL  | rename table event            |
     | drop index      | DDL  | drop index event              |
     | alter table     | DDL  | alter table event             |
+    | value range decrease | incompatible DDL  | 使列字段变小的 DDL 语句，如将 VARCHAR(20) 改为 VARCHAR(10) 的 ALTER TABLE MODIFY COLUMN 语句 |
+    | precision decrease | incompatible DDL  | 使列字段精度变小的 DDL 语句，如将 Decimal(10, 2) 改为 Decimal(10, 1) 的 ALTER TABLE MODIFY COLUMN 语句 |
+    | modify column | incompatible DDL  | 变更列字段类型的 DDL 语句，如将 INT 改为 VARCHAR 的 ALTER TABLE MODIFY COLUMN 语句 |
+    | rename | incompatible DDL  | 变更数据库名，列名和索引名的 DDL 语句，如 ALTER TABLE RENAME INDEX 语句 |
+    | drop | incompatible DDL  | 删除表中的列，索引，views 等的 DDL 语句，如 ALTER TABLE DROP COLUMN 语句 |
```
Comment (hfxsd) [inline dm/dm-binlog-event-filter.md:62]:
```suggestion
    | drop | incompatible DDL  | 删除表中的列、索引、views 等的 DDL 语句，如 ALTER TABLE DROP COLUMN 语句 |
```

File: dm/dm-binlog-event-filter.md [hunk 4]
```diff
@@ -54,18 +55,40 @@ filters:
...
     - `Ignore`：黑名单。如果满足下面两个条件之一就会被过滤掉：
         - 在该 rule 的 `events` 中。
         - 如果规则的 `sql-pattern` 不为空的话，对应的 SQL 可以匹配上 `sql-pattern` 中任意一项。
-    - 同一个表匹配上多个规则时，将会顺序应用这些规则，并且黑名单的优先级高于白名单，即如果同时存在规则 `Ignore` 和 `Do` 应用在某个表上，那么 `Ignore` 生效。
+    - `Error`：报错名单。如果满足下面两个条件之一就会报错：
+        - 在该 rule 的 `events` 中。
+        - 如果规则的 `sql-pattern` 不为空的话，对应的 SQL 可以匹配上 `sql-pattern` 中任意一项。
+    - 同一个表匹配上多个规则时，将会顺序应用这些规则，并且黑名单的优先级高于报错名单高于白名单，即如果同时存在规则 `Ignore` 和 `Error` 应用在某个表上，那么 `Ignore` 生效；如果同时存在规则 `Error` 和 `Do` 应用在某个表上，那么 `Error` 生效。
 
 ## 使用示例
```
Comment (hfxsd) [inline dm/dm-binlog-event-filter.md:93]: 可以加个 error 的示例吗

File: dm/dm-binlog-event-filter.md [hunk 5]
```diff
@@ -54,18 +55,40 @@ filters:
     | rename table    | DDL  | rename table event            |
     | drop index      | DDL  | drop index event              |
     | alter table     | DDL  | alter table event             |
+    | value range decrease | incompatible DDL  | 使列字段变小的 DDL 语句，如将 VARCHAR(20) 改为 VARCHAR(10) 的 ALTER TABLE MODIFY COLUMN 语句 |
+    | precision decrease | incompatible DDL  | 使列字段精度变小的 DDL 语句，如将 Decimal(10, 2) 改为 Decimal(10, 1) 的 ALTER TABLE MODIFY COLUMN 语句 |
+    | modify column | incompatible DDL  | 变更列字段类型的 DDL 语句，如将 INT 改为 VARCHAR 的 ALTER TABLE MODIFY COLUMN 语句 |
+    | rename | incompatible DDL  | 变更数据库名，列名和索引名的 DDL 语句，如 ALTER TABLE RENAME INDEX 语句 |
```
Comment (hfxsd) [inline dm/dm-binlog-event-filter.md:61]:
```suggestion
    | value range decrease | incompatible DDL  | 使列字段变小的 DDL 语句，如将 `VARCHAR(20)` 改为 `VARCHAR(10)` 的 `ALTER TABLE MODIFY COLUMN` 语句 |
    | precision decrease | incompatible DDL  | 使列字段精度变小的 DDL 语句，如将 `Decimal(10, 2)` 改为 `Decimal(10, 1)` 的 `ALTER TABLE MODIFY COLUMN` 语句 |
    | modify column | incompatible DDL  | 变更列字段类型的 DDL 语句，如将 `INT` 改为 `VARCHAR` 的 `ALTER TABLE MODIFY COLUMN` 语句 |
    | rename | incompatible DDL  | 变更数据库名，列名和索引名的 DDL 语句，如 `ALTER TABLE RENAME INDEX` 语句 |
```

File: dm/dm-binlog-event-filter.md [hunk 6]
```diff
@@ -54,18 +55,40 @@ filters:
...
+    | modify constraint | incompatible DDL  | 修改 constraint 的 DDL 语句，如 ALTER TABLE ADD CONSTRAINT 语句 |
+    | modify columns order | incompatible DDL  | 修改列顺序的 DDL 语句，如 ALTER TABLE CHANGE AFTER 语句 |
+    | modify charset | incompatible DDL  | 修改列 charset 的 DDL 语句，如 ALTER TABLE MODIFY CHARSET 语句 |
+    | modify collation | incompatible DDL  | 修改列 collation 的 DDL 语句，如 ALTER TABLE MODIFY COLLATE 语句 |
+    | remove auto increment | incompatible DDL  | 删除自增键的 DDL 语句 |
+    | modify storage engine | incompatible DDL  | 修改表存储引擎的 DDL 语句，如 ALTER TABLE ENGINE = MyISAM 语句 |
+    | reorganize partiton | incompatible DDL  | reorganize partiton 的 DDL 语句，如 ALTER TABLE REORGANIZE PARTITION 语句 |
+    | rebuild partiton | incompatible DDL  | rebuild partiton 的 DDL 语句，如 ALTER TABLE REBUILD PARTITION 语句 |
+    | exchange partiton | incompatible DDL  | exchange partiton 的 DDL 语句，如 ALTER TABLE EXCHANGE PARTITION 语句 |
+    | coalesce partiton | incompatible DDL  | coalesce partiton 的 DDL 语句，如 ALTER COALESCE PARTITION 语句 |
```
Comment (hfxsd) [inline dm/dm-binlog-event-filter.md:76]:
```suggestion
    | truncate | incompatible DDL  | truncate 表中的 partition 的 DDL 语句，如 `ALTER TABLE TRUNCATE PARTITION` 语句 |
    | modify pk | incompatible DDL  | 修改主键的 DDL 语句，如 `ALTER TABLE DROP PRIMARY KEY` 语句 |
    | modify uk | incompatible DDL  | 修改唯一键的 DDL 语句，如 `ALTER TABLE DROP UNIQUE KEY` 语句 |
    | modify default value | incompatible DDL  | 修改列默认值的 DDL 语句，如 `ALTER TABLE CHANGE DEFAULT` 语句 |
    | modify constraint | incompatible DDL  | 修改 constraint 的 DDL 语句，如 `ALTER TABLE ADD CONSTRAINT` 语句 |
    | modify columns order | incompatible DDL  | 修改列顺序的 DDL 语句，如 `ALTER TABLE CHANGE AFTER` 语句 |
    | modify charset | incompatible DDL  | 修改列 charset 的 DDL 语句，如 `ALTER TABLE MODIFY CHARSET` 语句 |
    | modify collation | incompatible DDL  | 修改列 collation 的 DDL 语句，如 `ALTER TABLE MODIFY COLLATE` 语句 |
    | remove auto increment | incompatible DDL  | 删除自增键的 DDL 语句 |
    | modify storage engine | incompatible DDL  | 修改表存储引擎的 DDL 语句，如 `ALTER TABLE ENGINE = MyISAM` 语句 |
    | reorganize partiton | incompatible DDL  | reorganize partiton 的 DDL 语句，如 `ALTER TABLE REORGANIZE PARTITION` 语句 |
    | rebuild partiton | incompatible DDL  | rebuild partiton 的 DDL 语句，如 `ALTER TABLE REBUILD PARTITION` 语句 |
    | exchange partiton | incompatible DDL  | exchange partiton 的 DDL 语句，如 `ALTER TABLE EXCHANGE PARTITION` 语句 |
    | coalesce partiton | incompatible DDL  | coalesce partiton 的 DDL 语句，如 `ALTER COALESCE PARTITION` 语句 |
```

File: dm/dm-binlog-event-filter.md [hunk 7]
```diff
@@ -54,18 +55,40 @@ filters:
     | rename table    | DDL  | rename table event            |
     | drop index      | DDL  | drop index event              |
     | alter table     | DDL  | alter table event             |
+    | value range decrease | incompatible DDL  | 使列字段变小的 DDL 语句，如将 `VARCHAR(20)` 改为 `VARCHAR(10)` 的 `ALTER TABLE MODIFY COLUMN` 语句 |
+    | precision decrease | incompatible DDL  | 使列字段精度变小的 DDL 语句，如将 `Decimal(10, 2)` 改为 `Decimal(10, 1)` 的 `ALTER TABLE MODIFY COLUMN` 语句 |
+    | modify column | incompatible DDL  | 变更列字段类型的 DDL 语句，如将 `INT` 改为 `VARCHAR` 的 `ALTER TABLE MODIFY COLUMN` 语句 |
+    | rename | incompatible DDL  | 变更数据库名，列名和索引名的 DDL 语句，如 `ALTER TABLE RENAME INDEX` 语句 |
+    | drop | incompatible DDL  | 删除表中的列、索引、views 等的 DDL 语句，如 ALTER TABLE DROP COLUMN 语句 |
```
Comment (hfxsd) [inline dm/dm-binlog-event-filter.md:62]:
```suggestion
    | drop | incompatible DDL  | 删除表中的列、索引、views 等的 DDL 语句，如 `ALTER TABLE DROP COLUMN` 语句 |
```

File: dm/dm-binlog-event-filter.md [hunk 8]
```diff
@@ -143,3 +168,16 @@ filters:
     sql-pattern: ["ALTER\\s+TABLE[\\s\\S]*ADD\\s+PARTITION", "ALTER\\s+TABLE[\\s\\S]*DROP\\s+PARTITION"]
     action: Ignore
 ```
+
+### 对部分 DDL 语句报错
+
+如需对部分上游业务的 DDL 语句报错，可采用如下设置
```
Comment (hfxsd) [inline dm/dm-binlog-event-filter.md:174]:
```suggestion
如需对部分上游业务的 DDL 语句报错，可采用如下设置：
```

File: dm/dm-binlog-event-filter.md [hunk 9]
```diff
@@ -39,33 +39,58 @@ filters:
...
         - 不在该 rule 的 `events` 中。
         - 如果规则的 `sql-pattern` 不为空的话，对应的 SQL 没有匹配上 `sql-pattern` 中任意一项。
     - `Ignore`：黑名单。如果满足下面两个条件之一就会被过滤掉：
         - 在该 rule 的 `events` 中。
         - 如果规则的 `sql-pattern` 不为空的话，对应的 SQL 可以匹配上 `sql-pattern` 中任意一项。
-    - 同一个表匹配上多个规则时，将会顺序应用这些规则，并且黑名单的优先级高于白名单，即如果同时存在规则 `Ignore` 和 `Do` 应用在某个表上，那么 `Ignore` 生效。
+    - `Error`：报错名单。如果满足下面两个条件之一就会报错：
+        - 在该 rule 的 `events` 中。
+        - 如果规则的 `sql-pattern` 不为空的话，对应的 SQL 可以匹配上 `sql-pattern` 中任意一项。
+    - 同一个表匹配上多个规则时，将会顺序应用这些规则，并且黑名单的优先级高于报错名单，报错名单的优先级高于白名单，即如果同时存在规则 `Ignore` 和 `Error` 应用在某个表上，那么 `Ignore` 生效；如果同时存在规则 `Error` 和 `Do` 应用在某个表上，那么 `Error` 生效。
```
Comment (hfxsd) [inline dm/dm-binlog-event-filter.md:93]:
```suggestion
    - 同一个表匹配上多个规则时，将会按顺序依次应用这些规则，并且黑名单的优先级高于报错名单，报错名单的优先级高于白名单，即如果同时存在规则 `Ignore` 和 `Error` 应用在某个表上，那么 `Ignore` 生效；如果同时存在规则 `Error` 和 `Do` 应用在某个表上，那么 `Error` 生效。
```

File: dm/dm-binlog-event-filter.md [hunk 10]
```diff
@@ -143,3 +168,16 @@ filters:
     sql-pattern: ["ALTER\\s+TABLE[\\s\\S]*ADD\\s+PARTITION", "ALTER\\s+TABLE[\\s\\S]*DROP\\s+PARTITION"]
     action: Ignore
 ```
+
+### 对部分 DDL 语句报错
+
+如需对部分上游业务的 DDL 语句报错，可采用如下设置：
```
Comment (Frank945946) [inline dm/dm-binlog-event-filter.md:174]:
```suggestion
如需对部分上游业务产生的 DDL 在 DM 同步到 TiDB 之前进行拦截并报错，可采用如下设置：
```

File: dm/dm-binlog-event-filter.md [hunk 11]
```diff
@@ -39,33 +39,58 @@ filters:
...
+    | value range decrease | incompatible DDL  | 使列字段变小的 DDL 语句，如将 `VARCHAR(20)` 改为 `VARCHAR(10)` 的 `ALTER TABLE MODIFY COLUMN` 语句 |
+    | precision decrease | incompatible DDL  | 使列字段精度变小的 DDL 语句，如将 `Decimal(10, 2)` 改为 `Decimal(10, 1)` 的 `ALTER TABLE MODIFY COLUMN` 语句 |
+    | modify column | incompatible DDL  | 变更列字段类型的 DDL 语句，如将 `INT` 改为 `VARCHAR` 的 `ALTER TABLE MODIFY COLUMN` 语句 |
+    | rename column | incompatible DDL  | 变更列名的 DDL 语句，如 `ALTER TABLE RENAME COLUMN` 语句 |
+    | rename index | incompatible DDL  | 变更索引名的 DDL 语句，如 `ALTER TABLE RENAME INDEX` 语句 |
+    | drop column | incompatible DDL  | 删除表中的列的 DDL 语句，如 `ALTER TABLE DROP COLUMN` 语句 |
+    | drop index | incompatible DDL  | 删除表中的索引的 DDL 语句，如 `ALTER TABLE DROP INDEX` 语句 |
+    | truncate table partition | incompatible DDL  | truncate 表中的 partition 的 DDL 语句，如 `ALTER TABLE TRUNCATE PARTITION` 语句 |
+    | drop primary key | incompatible DDL  | 修改主键的 DDL 语句，如 `ALTER TABLE DROP PRIMARY KEY` 语句 |
+    | drop unqiue key | incompatible DDL  | 修改唯一键的 DDL 语句，如 `ALTER TABLE DROP UNIQUE KEY` 语句 |
```
Comment (hfxsd) [inline dm/dm-binlog-event-filter.md:67]:
```suggestion
    | drop primary key | incompatible DDL  | 删除主键的 DDL 语句，如 `ALTER TABLE DROP PRIMARY KEY` 语句 |
    | drop unqiue key | incompatible DDL  |  删除唯一键的 DDL 语句，如 `ALTER TABLE DROP UNIQUE KEY` 语句 |
```

**Review by hfxsd — APPROVED** [review]

File: dm/dm-binlog-event-filter.md [hunk 12]
```diff
@@ -39,33 +39,58 @@ filters:
...
+    | drop primary key | incompatible DDL  | 删除主键的 DDL 语句，如 `ALTER TABLE DROP PRIMARY KEY` 语句 |
+    | drop unqiue key | incompatible DDL  |  删除唯一键的 DDL 语句，如 `ALTER TABLE DROP UNIQUE KEY` 语句 |
+    | modify default value | incompatible DDL  | 修改列默认值的 DDL 语句，如 `ALTER TABLE CHANGE DEFAULT` 语句 |
+    | modify constraint | incompatible DDL  | 修改 constraint 的 DDL 语句，如 `ALTER TABLE ADD CONSTRAINT` 语句 |
+    | modify columns order | incompatible DDL  | 修改列顺序的 DDL 语句，如 `ALTER TABLE CHANGE AFTER` 语句 |
+    | modify charset | incompatible DDL  | 修改列 charset 的 DDL 语句，如 `ALTER TABLE MODIFY CHARSET` 语句 |
+    | modify collation | incompatible DDL  | 修改列 collation 的 DDL 语句，如 `ALTER TABLE MODIFY COLLATE` 语句 |
+    | remove auto increment | incompatible DDL  | 删除自增键的 DDL 语句 |
+    | modify storage engine | incompatible DDL  | 修改表存储引擎的 DDL 语句，如 `ALTER TABLE ENGINE = MyISAM` 语句 |
+    | reorganize table partiton | incompatible DDL  | reorganize partiton 的 DDL 语句，如 `ALTER TABLE REORGANIZE PARTITION` 语句 |
```
Comment (qiancai) [inline dm/dm-binlog-event-filter.md:75]:
```suggestion
    | reorganize table partition | incompatible DDL  | 重组 partition 的 DDL 语句，如 `ALTER TABLE REORGANIZE PARTITION` 语句 |
```

File: dm/dm-binlog-event-filter.md [hunk 13]
```diff
@@ -39,33 +39,58 @@ filters:
...
+    | rebuild table partiton | incompatible DDL  | rebuild partiton 的 DDL 语句，如 `ALTER TABLE REBUILD PARTITION` 语句 |
+    | exchange table partiton | incompatible DDL  | exchange partiton 的 DDL 语句，如 `ALTER TABLE EXCHANGE PARTITION` 语句 |
```
Comment (qiancai) [inline dm/dm-binlog-event-filter.md:77]:
```suggestion
    | exchange table partition | incompatible DDL  | 交换分区的 DDL 语句，如 `ALTER TABLE EXCHANGE PARTITION` 语句 |
```

File: dm/dm-binlog-event-filter.md [hunk 14]
```diff
@@ -39,33 +39,58 @@ filters:
...
+    | drop unqiue key | incompatible DDL  |  删除唯一键的 DDL 语句，如 `ALTER TABLE DROP UNIQUE KEY` 语句 |
+    | modify default value | incompatible DDL  | 修改列默认值的 DDL 语句，如 `ALTER TABLE CHANGE DEFAULT` 语句 |
+    | modify constraint | incompatible DDL  | 修改 constraint 的 DDL 语句，如 `ALTER TABLE ADD CONSTRAINT` 语句 |
+    | modify columns order | incompatible DDL  | 修改列顺序的 DDL 语句，如 `ALTER TABLE CHANGE AFTER` 语句 |
+    | modify charset | incompatible DDL  | 修改列 charset 的 DDL 语句，如 `ALTER TABLE MODIFY CHARSET` 语句 |
+    | modify collation | incompatible DDL  | 修改列 collation 的 DDL 语句，如 `ALTER TABLE MODIFY COLLATE` 语句 |
+    | remove auto increment | incompatible DDL  | 删除自增键的 DDL 语句 |
+    | modify storage engine | incompatible DDL  | 修改表存储引擎的 DDL 语句，如 `ALTER TABLE ENGINE = MyISAM` 语句 |
+    | reorganize table partiton | incompatible DDL  | reorganize partiton 的 DDL 语句，如 `ALTER TABLE REORGANIZE PARTITION` 语句 |
+    | rebuild table partiton | incompatible DDL  | rebuild partiton 的 DDL 语句，如 `ALTER TABLE REBUILD PARTITION` 语句 |
```
Comment (qiancai) [inline dm/dm-binlog-event-filter.md:76]:
```suggestion
    | rebuild table partition | incompatible DDL  | 重建分区的 DDL 语句，如 `ALTER TABLE REBUILD PARTITION` 语句 |
```

File: dm/dm-binlog-event-filter.md [hunk 15]
```diff
@@ -39,33 +39,58 @@ filters:
...
+    | exchange table partiton | incompatible DDL  | exchange partiton 的 DDL 语句，如 `ALTER TABLE EXCHANGE PARTITION` 语句 |
+    | coalesce table partiton | incompatible DDL  | coalesce partiton 的 DDL 语句，如 `ALTER COALESCE PARTITION` 语句 |
 
 - `sql-pattern`：用于过滤指定的 DDL SQL 语句，支持正则表达式匹配，例如上面示例中的 `"^DROP\\s+PROCEDURE"`。
 
-- `action`：string (`Do` / `Ignore`)；进行下面规则判断，满足其中之一则过滤，否则不过滤。
+- `action`：string (`Do` / `Ignore`/ `Error`)；进行下面规则判断，满足其中之一则过滤或报错，否则不过滤。
```
Comment (qiancai) [inline dm/dm-binlog-event-filter.md:82]:
```suggestion
- `action`：string (`Do` / `Ignore`/ `Error`)；进行下面规则判断：
```

File: dm/dm-binlog-event-filter.md [hunk 16]
```diff
@@ -143,3 +168,16 @@ filters:
     sql-pattern: ["ALTER\\s+TABLE[\\s\\S]*ADD\\s+PARTITION", "ALTER\\s+TABLE[\\s\\S]*DROP\\s+PARTITION"]
     action: Ignore
 ```
+
+### 对部分 DDL 语句报错
+
+如需对部分上游业务产生的 DDL 语句在 DM 同步到 TiDB 之前进行拦截并报错，可采用如下设置：
```
Comment (qiancai) [inline dm/dm-binlog-event-filter.md:174]:
```suggestion
如需在 DM 同步上游业务数据到 TiDB 之前对部分 DDL 语句进行拦截并报错，可采用如下设置：
```

File: dm/dm-binlog-event-filter.md [hunk 17]
```diff
@@ -39,33 +39,58 @@ filters:
     | all             |      | 代表包含下面所有的 events        |
     | all dml         |      | 代表包含下面所有 DML events     |
     | all ddl         |      | 代表包含下面所有 DDL events     |
+    | incompatible ddl changes         |      | 代表包含下面所有 incompatible DDL events     |
```
[the remark this answers is not in the document]
Comment (GMHDBJD) [reply dm/dm-binlog-event-filter.md]: 可能导致数据丢失的 DDL（data loss），更新了

**Review by qiancai — APPROVED** [review]

File: dm/dm-binlog-event-filter.md [hunk 18]
```diff
@@ -39,33 +39,58 @@ filters:
...
+    | value range decrease | incompatible DDL  | 缩短列字段长度的 DDL 语句，如将 `VARCHAR(20)` 改为 `VARCHAR(10)` 的 `ALTER TABLE MODIFY COLUMN` 语句 |
+    | precision decrease | incompatible DDL  | 降低列字段精度的 DDL 语句，如将 `Decimal(10, 2)` 改为 `Decimal(10, 1)` 的 `ALTER TABLE MODIFY COLUMN` 语句 |
+    | modify column | incompatible DDL  | 变更列字段类型的 DDL 语句，如将 `INT` 改为 `VARCHAR` 的 `ALTER TABLE MODIFY COLUMN` 语句 |
+    | rename column | incompatible DDL  | 变更列名的 DDL 语句，如 `ALTER TABLE RENAME COLUMN` 语句 |
+    | rename index | incompatible DDL  | 变更索引名的 DDL 语句，如 `ALTER TABLE RENAME INDEX` 语句 |
+    | drop column | incompatible DDL  | 删除表中的列的 DDL 语句，如 `ALTER TABLE DROP COLUMN` 语句 |
+    | drop index | incompatible DDL  | 删除表中的索引的 DDL 语句，如 `ALTER TABLE DROP INDEX` 语句 |
+    | truncate table partition | incompatible DDL  | 清空表中指定分区的 DDL 语句，如 `ALTER TABLE TRUNCATE PARTITION` 语句 |
+    | drop primary key | incompatible DDL  | 删除主键的 DDL 语句，如 `ALTER TABLE DROP PRIMARY KEY` 语句 |
+    | drop unqiue key | incompatible DDL  |  删除唯一键的 DDL 语句，如 `ALTER TABLE DROP UNIQUE KEY` 语句 |
```
Comment (hfxsd) [inline dm/dm-binlog-event-filter.md:67]:
```suggestion
    | drop unique key | incompatible DDL  |  删除唯一键的 DDL 语句，如 `ALTER TABLE DROP UNIQUE KEY` 语句 |
```

**Pull request merged**